File: app/src/main/java/net/north101/android/ghplayertracker/data/PerkNote.java
package net.north101.android.ghplayertracker.data;

import org.parceler.Parcel;
import org.parceler.ParcelConstructor;

@Parcel
public class PerkNote {
    public static final int MAX_TICKS = 3;

    protected int ticks;

    @ParcelConstructor
    public PerkNote(int ticks) {
        this.setTicks(ticks);
    }

    public int getTicks() {
        return ticks;
    }

    public void setTicks(int ticks) {
        this.ticks = Math.min(Math.max(ticks, 0), 3);
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof PerkNote)) return false;

        PerkNote perkNote = (PerkNote) o;

        return getTicks() == perkNote.getTicks();
    }

    @Override
    public int hashCode() {
        return getTicks();
    }
}

File: app/src/main/java/net/north101/android/ghplayertracker/data/PerkNote.kt
package net.north101.android.ghplayertracker.data

import android.os.Parcelable
import kotlinx.android.parcel.IgnoredOnParcel
import kotlinx.android.parcel.Parcelize

@Parcelize
data class PerkNote(
        protected var _ticks: Int
) : Parcelable {
    @IgnoredOnParcel
    var ticks: Int
        get() = _ticks
        set(value) {
            _ticks = Math.min(Math.max(value, 0), MAX_TICKS)
        }

    companion object {
        const val MAX_TICKS = 3
    }
}
